CampusTour/AR/ARExplorerViewController.swift

```diff
@@ -0,0 +1,174 @@
+        self.view.addSubview(sceneView)
+        sceneView.snp.makeConstraints { $0.edges.equalToSuperview() }
+        
+        let backButton = UIButton() //TODO make this look better
+        backButton.setTitle("X", for: .normal)
+        backButton.backgroundColor = UIColor.red
+        backButton.layer.cornerRadius = 10
+        backButton.clipsToBounds = true
+        backButton.addTarget(self, action: #selector(closeArAndReturn), for: .touchUpInside)
+        self.view.addSubview(backButton)
+        backButton.snp.makeConstraints {
+            $0.trailing.equalToSuperview().inset(24)
+            $0.bottom.equalToSuperview().inset(24)
+        }
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        initializeAr()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        startAr()
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        stopAr()
+    }
+    
+    func initializeAr() {
+        AppDelegate.shared!.locationProvider.addLocationListener(repeats: true) { [weak self] currentLocation in
+            DispatchQueue.main.async {
+                self?.updateLocation(currentLocation: currentLocation)
+            }
+        }
+    }
+    
+    func updateLocation(currentLocation: CLLocation) {
+        for (i, info) in self.itemsOfInterestAndViews.enumerated() {
+            if info.node == nil { //initialize node
+                let planeWidth = CGFloat(2) //maximum width for the scene view in meters
+                let plane = SCNPlane(width: planeWidth,
+                                     height: planeWidth * (info.view.frame.height / info.view.frame.width))
+                plane.firstMaterial!.diffuse.contents = info.view.layer
+                plane.firstMaterial?.isDoubleSided = true
+                let itemNode = SCNNode(geometry: plane)
+                let displacement = ARGps.estimateDisplacement(from: currentLocation, to: info.item.location)
+                
+                itemNode.position = displacement
+                itemNode.constraints = [
+                    makeItemViewConstraint()
+                ]
+                self.sceneView.scene.rootNode.addChildNode(itemNode)
+                self.itemsOfInterestAndViews[i].node = itemNode
+            }
+            
+            if let camera = self.camera,
+                let node = info.node
+            {
+                let distance = (camera.transform.extractTranslation() - node.simdPosition).norm()
+                info.view.updateSubtitleWithDistance(meters: Double(distance))
+            }
+        }
+    }
+    
+    func startAr() {
+        let trackingConfig = ARWorldTrackingConfiguration()
+        trackingConfig.worldAlignment = .gravityAndHeading
+        sceneView.session.run(trackingConfig, options: [.resetTracking, .removeExistingAnchors])
+    }
+    
+    func stopAr() {
```

Comment:
Will there be sth here?

Reply:
there should be some cleanup, I haven't gotten to that yet